proposals/0296-increase-tx-size.md

```diff
@@ -0,0 +1,193 @@
+client as well as a frictional developer experience for many developers.
+
+A number of developers have also resorted to using Jito bundles to artificially
+increase the size of their transactions. This is a current workaround that
+developers are using, but it does not have the same guarantees as a transaction
+does on the protocol level, namely that bundles are not guaranteed to be atomic
+like singular transactions.
+
+For these reasons, increasing the current transaction size limit would enable
+developers to create new applications and use cases that were previously
+unfeasible.
+
+## New Terminology
+
+- 'v1 transaction' - A new transaction format that is designed to enable larger
+  transactions sizes while not having the address lookup table features 
+  introduced in v0 transactions.
+
+## Detailed Design
+
+Increasing the transaction size limit above the current MTU max size is
+possible with the use of QUIC. QUIC's RFC 9000 specification does not have an
+explicit maximum stream size, allowing for larger transactions to be sent.
+
+A new transaction format, `v1`, is proposed to enable larger transaction sizes.
+The `v1` transaction format would be:
+
+```
+VersionByte (u8) - >129 to distinguish from legacy/v0 formats
+LegacyHeader (u8, u8, u8) -- Required signatures from the current
+`MessageHeader` type
+Payload Length (u16) -- Total size excluding signatures
+NumInstructions (u8)
+TransactionConfigMask (u64) -- Bitmask of which config requests are present.
+ - 0: requested_compute_unit_limit
+ - 1: requested_compute_unit_price
+ - 2: requested_heap_size
+ - 3: requested_loaded_accounts_data_size_limit
+LifetimeSpecificier [u8; 32]
+NumAddresses (u8)
+Addresses [[u8; 32]] -- Number matches NumAddresses
+ConfigRequests [u64] -- Array of request values. (section size is popcount
+ TransactionConfigMask * 8). each value is a u64.
+Ixs [(u8, u8, u16, u16, u16)] -- Number matches NumInstructions. Values are 
+ (program_account_index, num_accounts, accounts_offset, num_data_bytes, 
+ bytes_offset)
+[TRAILING DATA SECTION] -- Length is such that PayloadLength is matched.
+Signatures [[u8; 64]] -- Length of `num_required_signatures` from
+ `LegacyHeader`
+```
+
+The [TRAILING DATA SECTION] contains instruction's serialized account indexes
+and instruction data bytes,
+in the order they appear in the transaction. Each instruction's account indexes
+must appear before its data bytes.
+For example, if there are two instructions:
+
+1. `Instruction { account_indexes: [0, 1], data: [2, 3, 4] }`
+2. `Instruction { account_indexes: [5, 6], data: [7, 8] }`
+
+The TRAILING DATA SECTION must look like this:
+
+```
+[0, 1, 5, 6, 2, 3, 4, 7, 8]
+```
+
+Lengths are serialized in the earlier `Ixs` section, and so are not repeated in
+the trailing data section.
+
+This new `v1` transaction format notably does not include address lookup
+tables.
+
+In consideration of what size the new transaction size limit should increase
```

Comment:
Everything from this line downwards shouldn't be in the details section IMO. The details should only include information needed for validator clients to implement the proposal.